curriculum/challenges/english/25-front-end-development/lab-availability-table/66b36358ed4f261d64840c24.md

```diff
@@ -0,0 +1,602 @@
+```
+
+You should define a `--solid-border` variable in your root selector.
+
+```js
+const root = new __helpers.CSSHelp(document).getStyle(':root');
+assert(!!root && root.getPropVal('--solid-border'));
+```
+
+You should define a `--dashed-border` variable in your root selector.
+
+```js
+const root = new __helpers.CSSHelp(document).getStyle(':root');
+assert(!!root && root.getPropVal('--dashed-border'));
+```
+
+You should target `td` elements that are children of `.sharp` elements.
+
+```js
+const tds = new __helpers.CSSHelp(document).getStyle('.sharp td') || new __helpers.CSSHelp(document).getStyle('.sharp>td');
+assert(tds);
+```
+
+You should use `--solid-border` to set the `bottom-border` of `td` elements that are children of `.sharp` elements.
+
+```js
+const tds = new __helpers.CSSHelp(document).getStyle('.sharp td') || new __helpers.CSSHelp(document).getStyle('.sharp>td');
+assert(tds && tds.borderBottom === 'var(--solid-border)');
+```
+
+You should target `td` elements that are children of `.half` elements.
+
+```js
+const tds = new __helpers.CSSHelp(document).getStyle('.half td') || new __helpers.CSSHelp(document).getStyle('.half>td');
+assert(tds);
+```
+
+You should use `--dashed-border` to set the `bottom-border` of `td` elements of `.half` elements.
+
+```js
+const tds = new __helpers.CSSHelp(document).getStyle('.half td') || new __helpers.CSSHelp(document).getStyle('.half>td');
+assert(tds && tds.borderBottom === 'var(--dashed-border)');
+```
+
+You should have a `div` element with the `id` of `legend`.
+
+```js
+assert.equal(document.querySelector('#legend')?.tagName, 'DIV');
+```
+
+You should have a `span` element with the text `Availability` inside your `#legend`.
+
+```js
+assert.equal(document.querySelector('#legend span')?.innerText, 'Availability');
+```
+
+You should have a `div` element with the `id` of `legend-gradient` inside your `#legend`.
+
+```js
+assert.equal(document.querySelector('#legend #legend-gradient')?.tagName, 'DIV');
+```
+
+You should set the background image of `#legend-gradient` to a linear gradient.
+
+```js
+const legendGradient = new __helpers.CSSHelp(document).getStyle('#legend-gradient');
+assert(!!legendGradient && legendGradient.backgroundImage.includes('linear-gradient('))
+```
+
+You should use two color-stops (expressed in percentage) to make the transition from one color to the following color a hard line for your `#legend-gradient`. Remember to use your `--color#` variables.
+
+```js
+const legendGradient = new __helpers.CSSHelp(document).getStyle('#legend-gradient');
+assert(!!legendGradient && legendGradient.backgroundImage.match(/var\(\s*--color[0-5]\s*\)\s+\d+%\s+\d+%/g).length === 6)
+```
+
+# --seed--
+
+## --seed-contents--
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Availability Table</title>
+    <link rel="stylesheet" href="styles.css">
```

Comment:
this will need to be removed per our discussion earlier

Reply:
We should also add a note like this:

```
**Note:** Be sure to link your stylesheet in your HTML and apply your CSS.
```